internal/util/funcutil/func.go

```diff
@@ -359,3 +361,42 @@ func IsGrpcErr(err error) bool {
 		err = errors.Unwrap(err)
 	}
 }
+
+func IsEmptyString(str string) bool {
+	return strings.TrimSpace(str) == ""
+}
+
+func HandleTenantForEtcdKey(prefix string, tenant string, key string) string {
+	res := prefix
+	if tenant != "" {
+		res += "/" + tenant
+	}
+	if key != "" {
+		res += "/" + key
+	}
+	return res
+}
+
+func IsRevoke(operateType milvuspb.OperatePrivilegeType) bool {
+	return operateType == milvuspb.OperatePrivilegeType_Revoke
+}
+
+func IsGrant(operateType milvuspb.OperatePrivilegeType) bool {
+	return operateType == milvuspb.OperatePrivilegeType_Grant
+}
+
+// IsKeyNotExistError Judging by the error message whether the key does not exist or not for the ectd server
+func IsKeyNotExistError(err error) bool {
+	return strings.Contains(err.Error(), "there is no value on key")
+}
+
+func EncodeUserRoleCache(user string, role string) string {
+	return fmt.Sprintf("%s/%s", user, role)
+}
+
+func DecodeUserRoleCache(cache string) (string, string) {
+	index := strings.LastIndex(cache, "/")
+	user := cache[:index]
+	role := cache[index+1:]
+	return user, role
+}
```

Comment:
if `cache` does not contains "/", this function will panic with out of bound error.

Reply:
yes. but `DecodeUserRoleCache` and `EncodeUserRoleCache` are a pair. so the check is ignored by me.

Reply:
Maybe we could not assume this input of `DecodeUserRoleCache` will always be output of `EncodeUserRoleCache`. It may be some noise of the env or future misuse. It's always better to prevent panicking if it's not wanted.

Reply:
yes, I want to improve it in the next pull request today. I will open the pull request about `rabc II`.